modules/proxy-certificates.adoc

```diff
@@ -0,0 +1,75 @@
+
+// Module included in the following assemblies:
+//
+// * authentication/understanding-certificates.adoc
+
+[id="proxy-certificates_{context}"]
+= Proxy certificates
+
+== Purpose
+
+The `trustedCA` field of `proxy` is a reference to a ConfigMap that contains a
+user-provided trusted certificate authority (CA) bundle. This bundle is merged
+with the Red Hat Enterprise Linux CoreOS (RHCOS) trust bundle and injected into
+the trust store of platform components that make egress HTTPS calls. For
+example, `image-registry-operator` calls an external image registry to download
+images. If `trustedCA` is not specified, only the RHCOS trust bundle is used for
+proxied HTTPS connections.
+
+// Managing the RHCOS trust bundle is outside the scope of proxy. I need to contact that team for those details.
```

Comment:
@miabbott Are you the correct content to help me build out these details for the customer?

Reply:
@ahardin-rh Yeah, I could help out there.  I don't claim to have all the answers, but I can help get them  :)

Reply:
@miabbott Thank you! Any help is appreciated! :smile:

Reply:
@miabbott Can you please provide details by end of week? We are needing the following in regard to the RHCOS trust bundle:

* The purpose
* File path
* Default expiration term
* How to set custom expiration term
* How to specify the expiration date of all certificates used in OpenShift when installing OpenShift
* Which services use it
* How to update/extend it
* Will certificates that are about to expire be automatically renewed by the operator?

Reply:
I'll do my best to answer these questions, but I don't have all the answers.

**Purpose**
Users want to provide custom CA certs to the RHCOS trust bundle so that they can use their own certificate infrastructure.

**File path**
Users will place CA certs in `/etc/pki/ca-trust/source/anchors`

**Default expiration term**
I believe the default expiration term should be defined by the CA certificate itself.  It is up to the CA admin to configure this for the certificate itself before it can be used by OCP/RHCOS.

**How to set custom expiration term**
Again I believe this is a function of the CA certificate itself.  It would be up to the CA admin to configure this for the certificate before it would be used by OCP/RHCOS

**Specify expiration date of all certs**
I believe this is outside the scope of writing CA certs to the RHCOS trust bundle.  At the very least, it is something that is not handled at the RHCOS/node level.

**Which services use it**
Presumably any service that is running on the RHCOS node would be able to use the trust bundle of the node.

**How to update/extend it**
The mechanism for writing CA certs to the RHCOS trust bundle is exactly the same as writing any other file to RHCOS, which is done through the use of MachineConfigs.  When the MCO applies the new MachineConfig that contains the new CA certs, the node will be rebooted.  During the next boot, the service `coreos-update-ca-trust.service` will run on the RHCOS nodes which will automatically update the trust bundle with the new CA certs.

An example MachineConfig that drops a CA cert on the worker nodes (note the value of the `source` field is not a real CA cert):

```yaml
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker
  name: 50-examplecorp-ca-cert
spec:
  config:
    ignition:
      version: 2.2.0
    storage:
      files:
      - contents:
          source: data:text/plain;charset=utf-8;base64,LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVORENDQXh5Z0F3SUJBZ0lKQU51bkkwRDY2MmNuTUEwR0NTcUdTSWIzRFFFQkN3VUFNSUdsTVFzd0NRWUQKV1FRR0V3SlZVekVYTUJVR0ExVUVDQXdPVG05eWRHZ2dRMkZ5YjJ4cGJtRXhFREFPQmdOVkJBY01CMUpoYkdWcApBMmd4RmpBVUJnTlZCQW9NRFZKbFpDQklZWFFzSUVsdVl5NHhFekFSQmdOVkJBc01DbEpsWkNCSVlYUWdTVlF4Ckh6QVpCZ05WQkFNTUVsSmxaQ0JJWVhRZ1NWUWdVbTl2ZENCRFFURWhNQjhHQ1NxR1NJYjNEUUVKQVJZU2FXNW0KWGpDQnBURUxNQWtHQTFVRUJoTUNWVk14RnpBVkJnTlZCQWdNRGs1dmNuUm9JRU5oY205c2FXNWhNUkF3RGdZRApXUVFIREFkU1lXeGxhV2RvTVJZd0ZBWURWUVFLREExU1pXUWdTR0YwTENCSmJtTXVNUk13RVFZRFZRUUxEQXBTCkFXUWdTR0YwSUVsVU1Sc3dHUVlEVlFRRERCSlNaV1FnU0dGMElFbFVJRkp2YjNRZ1EwRXhJVEFmQmdrcWhraUcKMHcwQkNRRVdFbWx1Wm05elpXTkFjbVZrYUdGMExtTnZiVENDQVNJd0RRWUpLb1pJaHZjTkFRRUJCUUFEZ2dFUApCRENDQVFvQ2dnRUJBTFF0OU9KUWg2R0M1TFQxZzgwcU5oMHU1MEJRNHNaL3laOGFFVHh0KzVsblBWWDZNSEt6CmQvaTdsRHFUZlRjZkxMMm55VUJkMmZRRGsxQjBmeHJza2hHSUlaM2lmUDFQczRsdFRrdjhoUlNvYjNWdE5xU28KSHhrS2Z2RDJQS2pUUHhEUFdZeXJ1eTlpckxaaW9NZmZpM2kvZ0N1dDBaV3RBeU8zTVZINXFXRi9lbkt3Z1BFUwpZOXBvK1RkQ3ZSQi9SVU9iQmFNNzYxRWNyTFNNMUdxSE51ZVNmcW5obzNBakxRNmRCblBXbG82MzhabTFWZWJLCkNFTHloa0xXTVNGa0t3RG1uZTBqUTAyWTRnMDc1dkNLdkNzQ0F3RUFBYU5qTUdFd0hRWURWUjBPQkJZRUZIN1IKNXlDK1VlaElJUGV1TDhacXczUHpiZ2NaTUI4R0ExVWRJd1FZTUJhQUZIN1I0eUMrVWVoSUlQZXVMOFpxdzNQegpjZ2NaTUE4R0ExVWRFd0VCL3dRRk1BTUJBZjh3RGdZRFZSMFBBUUgvQkFRREFnR0dNQTBHQ1NxR1NJYjNEUUVCCkR3VUFBNElCQVFCRE52RDJWbTlzQTVBOUFsT0pSOCtlbjVYejloWGN4SkI1cGh4Y1pROGpGb0cwNFZzaHZkMGUKTUVuVXJNY2ZGZ0laNG5qTUtUUUNNNFpGVVBBaWV5THg0ZjUySHVEb3BwM2U1SnlJTWZXK0tGY05JcEt3Q3NhawpwU29LdElVT3NVSks3cUJWWnhjckl5ZVFWMnFjWU9lWmh0UzV3QnFJd09BaEZ3bENFVDdaZTU4UUhtUzQ4c2xqCjVlVGtSaml2QWxFeHJGektjbGpDNGF4S1Fsbk92VkF6eitHbTMyVTB4UEJGNEJ5ZVBWeENKVUh3MVRzeVRtZWwKU3hORXA3eUhvWGN3bitmWG5hK3Q1SldoMWd4VVp0eTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
        filesystem: root
        mode: 0644
        path: /etc/pki/ca-trust/source/anchors/examplecorp-ca-cert
```

**Automatic renewal by the Operator**
I'm not aware of any operator that can auto-renew certificates on the RHCOS nodes

---
For additional details about the management of CA certs on the nodes and/or the `coreos-update-ca-trust.service`, please see the following:

https://github.com/openshift/machine-config-operator/issues/528
https://github.com/coreos/fedora-coreos-config/pull/120

Reply:
@miabbott Thank you!